Makefile

```diff
@@ -1,252 +1,62 @@
-	find client assets \
-		-type f \
-		-name '*.scss' \
-)
-JS_FILES := $(shell \
-	find . \
-		-not \( -path './.git' -prune \) \
-		-not \( -path './build' -prune \) \
-		-not \( -path './node_modules' -prune \) \
-		-not \( -path './public' -prune \) \
-		-type f \
-		\( -name '*.js' -or -name '*.jsx' \) \
-)
-# same as JS_FILES, but excludes extensions
-JS_FILES_FOR_TRANSLATE := $(shell \
-	find . \
-		-not \( -path './.git' -prune \) \
-		-not \( -path './build' -prune \) \
-		-not \( -path './node_modules' -prune \) \
-		-not \( -path './public' -prune \) \
-		-not \( -path './client/extensions' -prune \) \
-		-type f \
-		\( -name '*.js' -or -name '*.jsx' \) \
-)
-MD_FILES := $(shell \
-	find . \
-		-not \( -path './.git' -prune \) \
-		-not \( -path './build' -prune \) \
-		-not \( -path './node_modules' -prune \) \
-		-not \( -path './public' -prune \) \
-		-type f \
-		-name '*.md' \
-	| sed 's/ /\\ /g' \
-)
-COMPONENTS_USAGE_STATS_FILES = $(shell \
-	find client \
-		-not \( -path '*/docs/*' -prune \) \
-		-not \( -path '*/test/*' -prune \) \
-		-not \( -path '*/docs-example/*' -prune \) \
-		-type f \
-		\( -name '*.js' -or -name '*.jsx' \) \
-)
-COMPONENTS_PROPTYPE_FILES = $(shell \
-	find client \
-		-name 'index.jsx' \
-		-or -name 'index.js' \
-		-or -name 'example.jsx' \
-		-and -not -path '*/test/*' \
-)
-
-CONFIG_FILES = $(shell \
-	find config \
-		-name '*.json' \
-)
-
-
-# variables
-NODE_ENV ?= development
-CALYPSO_ENV ?= $(NODE_ENV)
-
-export NODE_ENV := $(NODE_ENV)
-export CALYPSO_ENV := $(CALYPSO_ENV)
-export NODE_PATH := server$(SEPARATOR)client$(SEPARATOR).
-
 .DEFAULT_GOAL := install
 
-welcome:
-	@printf "\033[36m             _                           \n"
-	@printf "\033[36m    ___ __ _| |_   _ _ __  ___  ___      \n"
-	@printf "\033[36m   / __/ _\` | | | | | '_ \/ __|/ _ \\   \n"
-	@printf "\033[36m  | (_| (_| | | |_| | |_) \__ \ (_) |    \n"
-	@printf "\033[36m   \___\__,_|_|\__, | .__/|___/\___/     \n"
-	@printf "\033[36m               |___/|_|                  \n"
-	@printf "\033[m\n"
-
-install: node_modules
-
-# Simply running `make run` will spawn the Node.js server instance.
-run: welcome githooks install build
-	@$(NODE) $(NODE_ARGS) build/bundle.js
-
-dashboard: install
-	@$(NODE_BIN)/webpack-dashboard -- make run
-
-# a helper rule to ensure that a specific module is installed,
-# without relying on a generic `npm install` command
-node_modules/%:
-	@$(NPM) install $(notdir $@)
-
-node-version: node_modules/semver
-	@$(BIN)/check-node-version
-
-# ensures that the `node_modules` directory is installed and up-to-date with
-# the dependencies listed in the "package.json" file.
-node_modules: package.json | node-version
-	@$(NPM) prune
-	@$(NPM) install
-	@touch node_modules
-
-test: build
-	@$(NPM) test
+install:
+	npm run install-if-needed
 
-lint: node_modules/eslint node_modules/eslint-plugin-react node_modules/babel-eslint config-defaults-lint mixedindentlint
-	@$(NPM) run lint
+run:
+	npm start
```

Comment:
No, not intentionally, I didn't even know what it was for. It makes them less verbose, so I've added it to every command.